src/robusta/core/sinks/pagerduty/pagerduty_sink.py

```diff
@@ -0,0 +1,119 @@
+        self.url = sink_config.pagerduty_sink.url        
+        self.api_key = sink_config.pagerduty_sink.api_key        
+    
+    def __to_pagerduty_severity_type(severity: FindingSeverity):
+        # must be one of [critical, error, warning, info]
+        # Default Incident Urgency is interpreted as [HIGH, HIGH, LOW, LOW]
+        # https://support.pagerduty.com/docs/dynamic-notifications
+        if severity == FindingSeverity.HIGH:
+            return "critical"
+        elif severity == FindingSeverity.MEDIUM:
+            return "error"        
+        elif severity == FindingSeverity.LOW:
+            return "warning"
+        elif severity == FindingSeverity.INFO:
+            return "info"        
+        elif severity == FindingSeverity.DEBUG:
+            return "info"      
+        else:
+            return "critical"
+        
+    def __to_pagerduty_status_type(title: str):
+        # very dirty implementation, I am deeply sorry
+        # must be one of [trigger, acknowledge or resolve]
+        if title.startswith('[RESOLVED]'):
+            return "resolve"
+        else:
+            return "trigger"
+
+    def write_finding(self, finding: Finding, platform_enabled: bool):
+        custom_details: dict = {}
+
+        if platform_enabled:
+            custom_details["🔎 Investigate"] = finding.investigate_uri
+
+            if finding.add_silence_url:
+                custom_details[
+                    "🔕 Silence"
+                ] = finding.get_prometheus_silence_url(self.cluster_name)
+
+        # custom fields
+        custom_details["Resource"] = finding.subject.name
+        custom_details["Source"] = self.cluster_name
+        custom_details["Namespace"] = finding.subject.namespace
+        custom_details["Node"] = finding.subject.node        
+        custom_details["Monitoring Tool"] = str(finding.source.name)
+        custom_details["Severity"] = PagerdutySink.__to_pagerduty_severity_type(finding.severity).upper()
+        custom_details["ID"] = finding.fingerprint
+        custom_details["Description"] = finding.description
+        custom_details["Caption"] = f"{finding.severity.to_emoji()} {PagerdutySink.__to_pagerduty_severity_type(finding.severity)} - {finding.title}"
+
+        message_lines = ""
+        if finding.description:
+            message_lines = finding.description + "\n\n"
+
+        for enrichment in finding.enrichments:
+            for block in enrichment.blocks:
+                text = self.__to_unformatted_text(block)
+                if not text:
+                    continue
+
+                message_lines += text + "\n\n"
+        
+        custom_details["state_message"] = message_lines
+
+        body = {
+            "payload": {
+                "summary": finding.title,        
+                "severity": PagerdutySink.__to_pagerduty_severity_type(finding.severity),
+                "source": str(finding.source.name),                                
```

Comment:
Would it make more sense to set `source` to something more specific? Right now, the values for `finding.source.name` will almost always be "kubernetes_api_server" or "prometheus" which are somewhat broad.

For example, we could map the following way:

* `source` is set to the cluster name
* `component` is set to the exact resource within the cluster that is associated with the alert - i.e.`finding.subject` [1]
* `group` is set to the resource's high-level component - i.e. `finding.service_key`

To give a concrete example, when an alert occurs on a pod named `some-deployment-abcd-efgh` then:
`component="some_namespace/pod/some-deployment-abcd-efgh"`
`group="some_namespace/deployment/some-deployment"`

The motivation here is to expose enough data to PagerDuty that you'll have the full flexibility to group however you want on the PagerDuty end. That said, I don't have much experience with PagerDuty, so let me know if this even makes sense or if I'm missing something obvious!

[1] To easily implement this, we should really add on a `FindingSubject.__str__` method to easily convert the finding subject to a string in a consistent manner - e.g. `{namespace}/{resource_type}/{name}`

Reply:
Source could be problematic when you are piping alerts from Global Prometheus (company level Prometheus) to Robusta and not from local (cluster level Prometheus) but I will do the change nevertheless, can always change back :)

Reply:
@taivo123 Oh, got it. It's possible to also send:

Global Prometheus -> Robusta Relay -> Relevant Robusta cluster -> PagerDuty

In which case the cluster_name will be correct.

If you need to send direct we can also look at other options or exposing a flag for this later.

Thanks for explaining.